cassandra-diagnostics-connector21/src/main/java/io/smartcat/cassandra/diagnostics/connector/DiagnosticsQueryHandler.java

```diff
@@ -0,0 +1,125 @@
+package io.smartcat.cassandra.diagnostics.connector;
+
+import org.apache.cassandra.cql3.BatchQueryOptions;
+import org.apache.cassandra.cql3.CQLStatement;
+import org.apache.cassandra.cql3.QueryHandler;
+import org.apache.cassandra.cql3.QueryOptions;
+import org.apache.cassandra.cql3.QueryProcessor;
+import org.apache.cassandra.cql3.statements.BatchStatement;
+import org.apache.cassandra.cql3.statements.ParsedStatement;
+import org.apache.cassandra.exceptions.InvalidRequestException;
+import org.apache.cassandra.exceptions.RequestExecutionException;
+import org.apache.cassandra.exceptions.RequestValidationException;
+import org.apache.cassandra.service.QueryState;
+import org.apache.cassandra.transport.messages.ResultMessage;
+import org.apache.cassandra.transport.messages.ResultMessage.Prepared;
+import org.apache.cassandra.utils.MD5Digest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.smartcat.cassandra.diagnostics.Diagnostics;
+
+/**
+ * Custom diagnostics query handler which hooks into query path and reports queries.
+ */
+public class DiagnosticsQueryHandler implements QueryHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DiagnosticsQueryHandler.class);
+
+    private static final String INITIALIZATION_THREAD_NAME = "cassandra-diagnostics-agent";
+
+    private final QueryProcessor queryProcessor = QueryProcessor.instance;
+
+    private static Connector21QueryReporter queryReporter;
```

Comment:
Well my idea was to have one step towards query reporter not tied to Cassandra and extract it to commons. To do that we need to take out CQLStatement part and ClientState and send only plain values to reporter (strings, longs etc.).

This is why I went this way and left this part to following refactoring not to change to much.